lib/std/debug.zig

```diff
@@ -280,22 +280,31 @@ pub const StackIterator = struct {
         };
     }
 
+    // On some architectures such as x86 the frame pointer is the address where
+    // the previous fp is stored, while on some other architectures such as
+    // RISC-V it points to the "top" of the frame, just above where the previous
+    // fp and the return address are stored.
+    const fp_adjust_factor = if (builtin.arch == .riscv32 or builtin.arch == .riscv64)
+        2 * @sizeOf(usize)
+    else
+        0;
+
     fn next(self: *StackIterator) ?usize {
-        if (self.fp == 0) return null;
-        self.fp = @intToPtr(*const usize, self.fp).*;
-        if (self.fp == 0) return null;
+        if (self.fp < fp_adjust_factor) return null;
```

Comment:
`usize(self.fp) < 0` always false on non riscv

Reply:
Good catch, turning the `<` into `<=` should not do any harm (beside excluding a very edge-case where `s0` is exactly 16... but that'd be really strange).

@andrewrk wanna do this change or should I send another PR?

Reply:
PR would be appreciated as I'm travelling this weekend, back home Tuesday.